kimu/core/explode_lines.py

```diff
@@ -0,0 +1,61 @@
+from qgis import processing
+from qgis.core import (
+    QgsFeatureRequest,
+    QgsProcessingFeatureSourceDefinition,
+    QgsProject,
+    QgsVectorLayer,
+    QgsWkbTypes,
+)
+from qgis.utils import iface
+
+from ..qgis_plugin_tools.tools.custom_logging import setup_logger
+from ..qgis_plugin_tools.tools.i18n import tr
+from ..qgis_plugin_tools.tools.resources import plugin_name
+
+LOGGER = setup_logger(plugin_name())
+
+
+class ExplodeLines:
```

Comment:
Not related to the line intersection functionality, but may be an useful tool. Ideally this would be its own branch/PR, but probably not worth the effort, IMO it's fine here.

Reply:
Keep this here, as it's useful for the current implementation (see comment in line intersection tool)

Reply:
I did as you suggested and modified the Explode Lines tool so, that it produces a new line layer instead of a point layer. I still did not want to give up on this explode lines to points functionality since I fear we need to create a version for determining the intersection point from a point layer and in that case we probably need this functionality.. So now this PR contains 3 new buttons..